clinica/pipelines/t1_freesurfer_longitudinal/t1_freesurfer_template_utils.py

```diff
@@ -0,0 +1,1323 @@
+            error_msg, caps_dir)
+        error_msg = '{0} Cannot safely remove.'.format(error_msg)
+        raise IOError(error_msg)
+
+    # remove the caps target
+    if os.path.exists(caps_target_real):
+        shutil.rmtree(caps_target_real)
+    else:
+        error_msg = 'Error: folder {0} does not exist'.format(caps_target)
+        raise IOError(error_msg)
+
+
+
+def copy_to_caps(
+        in_subject,
+        in_subject_dir,
+        in_caps_target,
+        in_caps_dir,
+        in_overwrite_caps,
+        in_template_created):
+    """Copy template folder to CAPS dir
+
+    Move the template folder created by recon-all base to its correct
+    location in the CAPS tree structure.
+
+    Args:
+        in_subject (string): (unique) participant_id
+        in_subject_dir (string): location of the template folder (in the
+            working directory)
+        in_caps_target (string): path to CAPS target subfolder for the
+            template connected to the current subject
+        in_caps_dir (string): CAPS directory to contain the output
+            Used to make sure the folder we are overwriting (if
+            overwrite-caps option provided) are subolders of the input
+            CAPS dir
+        in_overwrite_caps (string): Option provided by the user to state
+            whether the data in the CAPS folder (i.e.,
+            previously-computed template) should be overwritten or not
+            by a new run. By default, do not overwrite. Will overwrite
+            for the following values: 'true', 'True' and 'TRUE'
+        in_template_created (Boolean): Used to force template
+            recon-all -base to be run before moving anything to the CAPS
+            directory
+
+    Returns:
+        out_copy2_caps (boolean): flag to indicate whether or not the
+            copy was successful
+    """
+    import os
+    import shutil
+    import clinica.pipelines.t1_freesurfer_longitudinal.t1_freesurfer_template_utils as utils
+
+    # Convert overwrite-caps flag to boolean
+    force_overwrite = utils.boolean_overwrite_caps(in_overwrite_caps)
+
+    # define template location in the working directory
+    # (repeating twice in_subject because in_subject_dir/in_subject
+    # contains both in_subject template and cross-sectional
+    # symlinks)
+    wd_template_location = os.path.join(in_subject_dir, in_subject, in_subject)
+
+    # Check if overwrite-caps option provided. If provided, delete the
+    # target subdirectory in CAPS folder
+    if force_overwrite:
+        # check the target subdirectory exists in CAPS
+        if os.path.exists(in_caps_target):
+            # remove
+            utils.safer_rmtree(in_caps_target, in_caps_dir)
+
+    # copy the template from working directory to CAPS directory
+    shutil.copytree(wd_template_location, in_caps_target)
+    out_copy2_caps = True
+
+    return out_copy2_caps
+
+
+
+def sendto_longcorr(
+        in_copy2_caps,
```

Comment:
I added this variable to force node '4_copy2caps' (utils: copy_to_caps) to be run before node '5_sendto_longcorr' (utils: sendto_longcorr), so that we are sure the template creation pipeline is done before sending any data over to the longitudinal-correction pipeline.

If there are any other mechanisms to force node 1 to be run before node 2 without having to connect output variables of node 1 to input variables of node 2 I'm willing to use those instead.